emu/src/cpu/arm7tdmi.rs

```diff
@@ -1019,6 +1019,24 @@ impl Arm7tdmi {
         Some(SIZE_OF_THUMB_INSTRUCTION)
     }
 
+    fn load_address(&mut self, op_code: ThumbModeOpcode) -> Option<u32> {
+        let sp = op_code.get_bit(11);
+        let rd: usize = op_code.get_bits(8..=10).try_into().unwrap();
+        let word8: u32 = op_code.get_bits(0..=7).into();
+        let word8 = word8 << 2;
+
+        let v = if sp {
+            let stack_pointer = self.registers.register_at(REG_SP);
+            stack_pointer + word8
```

Comment:
I think `wrapping_add` would be safer here (also 2 lines down)

Reply:
Nice